### PR TITLE
Refactored gradle to fix bug that was causing CQL to crash

### LINKED ISSUE #177: Check CqlEvaluatorSampleApp & Integrate CQL Library as demo app into FHIR core
This is a simple app that creates a cql-evaluator and runs the WHO ANC Recommendation A2 content. It was created using Android Studio. Link to [repo](https://github.com/DBCG/CqlEvaluatorSampleApp)

Checks 
- [x] Check the structure of the FHIR resources being loaded from the assets folder 
- [x] Test functionality of loading and execution of CQL on Android client
- [x] Check whether $measure_evaluate is being run locally and the related functions on the sample app
- [x] Document the finding on this issue/ticket
 

Sample Screenshots

.  

Integration
- [x] Integrate it into FHIR core as a demo app for mADX  with some enhancements which include 
- [ ] Removed reference FHIR resources from Assets folder on the app and instead sync them from HAPI FHIR Server
- [x] List Patients from the server and not assets folder
- [ ] Add screen for Selection of patient and definition to generate the operation definition 

After this we  proceed to
- [ ] #404 
- [ ] #405 
- [ ] Run $measure_evaluate on server side and then later locally on Android client using the Workflow API

### PR DESCRIPTION
**Pull request by kimigxfoxy**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #177

**Type**
Choose one: (Bug fix) 

**Checklist**
- [x] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [x] I have added any strings visible on UI components to the `strings.xml` file
- [x] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [x] I have built and run the fhircore app to verify my change fixes the issue and/or does not break the app

### CODE REVIEW

File: android/engine/build.gradle [hunk 1]
```diff
@@ -102,17 +105,23 @@ dependencies {
...
-    implementation 'org.opencds.cqf.cql:evaluator:' + cqlEvaluatorVersion
+    implementation group: 'xerces', name: 'xercesImpl', version: '2.11.0'
+    implementation(group: "com.github.java-json-tools", name: "msg-simple", version: "1.2");
+    implementation(group: "com.github.java-json-tools", name: "jackson-coreutils", version: "2.0");
+    implementation 'com.fasterxml.jackson.core:jackson-core:2.12.2'
+    implementation 'com.fasterxml.jackson.core:jackson-databind:2.12.2'
+    implementation 'com.fasterxml.jackson.core:jackson-annotations:2.12.2'
+    implementation 'org.codehaus.woodstox:woodstox-core-asl:4.4.1'
+    implementation 'javax.xml.stream:stax-api:1.0-2'
+    implementation "ca.uhn.hapi.fhir:hapi-fhir-android:5.4.0"
```
Comment (ndegwamartin) [inline android/engine/build.gradle:116]: Maybe check step by step which of these libraries we either already have or don't need for the CQL evaluation feature to work

e.g. you can run `./gradlew :engine:dependencies` command from the **android** folder in fhircore

Comment (kimigxfoxy) [reply android/engine/build.gradle:116]: @ndegwamartin The libraries added are all need for CQL Evaluation especially during runtime

Comment (ndegwamartin) [reply android/engine/build.gradle:116]: @kimigxfoxy for example when I run the above shared command i.e. `./gradlew :engine:dependencies` I see that the jackson dependencies already exist in the dependency tree?

Comment (kimigxfoxy) [reply android/engine/build.gradle]: @ndegwamartin I have removed the Jackson dependencies in the CQL section and tested that CQL evaluate in app works

**Review by ndegwamartin — APPROVED** [review]

**Pull request merged**